crates/ditto-codegen-js/tests/golden/match_expressions.js

```diff
@@ -8,6 +8,20 @@ function ManyFields($0, $1, $2, $3) {
   return ["ManyFields", $0, $1, $2, $3];
 }
 const Nothing = ["Nothing"];
+function nested_pattern(m) {
+  if (
+    m[0] === "Just" &&
```

Comment:
previously this was being generated in the reverse order, which meant a `Nothing` argument would blow up because it's not legit to call `[1][1][1][0]` on `["Nothing"]`.